lib/msprime.c

```diff
@@ -2116,7 +2126,8 @@ alloc_individual(individual_t *ind, size_t ploidy)
     // Better to allocate these as a block?
     ind->parents = malloc(ploidy * sizeof(individual_t *));
     ind->segments = malloc(ploidy * sizeof(avl_tree_t));
-    if (ind->parents == NULL || ind->segments == NULL) {
+    ind->nodes = malloc(ploidy * sizeof(avl_node_t *));
```

Comment:
This is the wrong size - should be ``sizeof(tsk_id_t)`` or (better) ``sizeof(*ind->nodes)``, so you don't have to write down what the type is twice.